src/screens/settings/MainSettingsScreen.tsx

```diff
@@ -54,14 +56,38 @@ export default class MainSettingsScreen
     }
 
     public componentWillMount(): void {
-        // TODO proper state handling
+        // TODO properly load schedules from disk
         this.setState({ schedules: testSchedulesList });
     }
 
+    public onScheduleItemToggled(key: number, newEnabled: boolean): void {
+        const newSchedules: MainSettingsScreenSchedulesState[] = this.state.schedules;
+        newSchedules[key].enabled = newEnabled;
+
+        if (newEnabled) {
+            newSchedules.forEach((s: MainSettingsScreenSchedulesState) => {
+                if (s.key !== key) {
+                    s.enabled = false;
+                    s.ref.forceEnabled(false);
+                }
+            });
+        }
+
+        this.setState({ schedules: newSchedules });
+    }
+
     public render(): ReactNode {
         return (
             <SectionList
-                renderItem={({item}) => <ScheduleListItem title={item.name} enabled={item.enabled} />}
+                renderItem={({ item }) => (
+                    <ScheduleListItem
+                        ref={(me: ScheduleListItem) => { this.state.schedules[item.key].ref = me; }}
+                        onSwitchToggled={this.onScheduleItemToggled.bind(this)}
```

Comment:
Slight suggestion here: Instead of having a separate `myKey` parameter, you can do:
```typescript
this.onScheduleItemToggled.bind(this, item.key)
```
which will bind the key to `onScheduleItemToggled` and return a new function that only has `(enabled: boolean): void` as its signature.

That way we aren't having to tell the item component anything about its parent.

Reply:
Oh, you can make suggestions inline. That's cool.

Reply:
Will take a look at this. I don't quite understand the syntax you're suggesting, but I do like having components be as naïve as possible.

Reply:
Just took a look at [the MDN docs on `Function.prototype.bind()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind), that's dope! Changing now.